src/charm.py

```diff
@@ -245,22 +251,31 @@ def _get_s3_env(self) -> typing.Dict[str, typing.Any]:
 
         return s3_env
 
-    def _create_discourse_environment_settings(self) -> typing.Dict[str, typing.Any]:
-        """Create a layer config based on our current configuration.
+    def _get_redis_relation_data(self) -> typing.Tuple[typing.Any, typing.Any]:
+        """Get the hostname and port from the redis relation data.
 
         Returns:
-            Dictionary with all the environment settings.
+            Tuple with the hostname and port of the related redis
         """
-        # Get redis connection information from the relation.
-        redis_hostname = None
-        redis_port = 6379
         # This is the current recommended way of accessing the relation data.
         for redis_unit in self._stored.redis_relation:  # type: ignore
```

Comment:
Hmm this doesn't look great, I guess we should moved away from using `_stored`

Reply:
[We can't](https://github.com/canonical/redis-k8s-operator/issues/74) for now.